### DISCUSSION TITLE
How implement an internal state change

### DISCUSSION THREAD

**Discussion by acquleo** [dotnet-state-machine/stateless#542] [Q&A] [answered]
Hi,
I've got the following requirement.
Given the following state of the management of an alarm
Inactive acknowledged
Inactive not acknowledged
Active acknowledged
Active not acknowledged 

And the following triggers
Activate
Deactivate
Acknowledged

When I'm in the state inactive not acknowledged
A trigger activate is triggered

The final state would be 
Active not acknowledged

But because an internal behavior we need to perform an automatic acknowledge and change the state to:
Inactive acknowledged
Before go to the final state

How can I implement this scenario?

Thanks
Leonardo

**Comment by HenningNT** [comment, marked as answer]
Looks like you need something like this?

![image](https://github.com/dotnet-state-machine/stateless/assets/13196280/cdf94435-2826-49d3-a9be-2a6c97a5122a)
